### ISSUE TITLE
FlexiMerge and FlexiRoute are not flexible enough

### ISSUE THREAD

**Issue by patriknw** [akka/akka#16753]
It should be possible to emit more than one element. Perhaps we could enter an internal emitting state under the hood and wake up user state only after all elements have been emitted.

It should be possible to emit from the completion handler, before finishing.

**Comment by jrudolph** [comment]
+1 I've had to add post-tranformer stages for both merges and routes to be able to customize closing behavior and inject elements before the tear down.

**Comment by williamleferrand** [comment]
+ 1, we have merging code that blend n streams that we can't port to 1.0-RC2. Any update on this issue? Many thanks!

**Comment by drewhk** [comment]
This will be fixed only after 1.0

**Comment by williamleferrand** [comment]
Thanks for the update; we'll work around it!

**Comment by adamw** [comment]
+1

I have a use-case where demand is "accumulated": if there is demand and an element is read from an input, the fleximerge state may decide not to emit anything (put the element aside to emit it later); especially then emitting multiple elements (equal to the accumulated demand) should be possible.

**Comment by drewhk** [comment]
Yes, we need to solve this. We want 1.0 out of the door first though.

**Comment by patriknw** [comment]
I heard more interest in this feature.

**Comment by OlafurPa** [comment]
+1

I need to merge two streams and accumulate a list of elements to be emitted at some later time. Being able to emit multiple elements and also being able to emit them from the completion handler is really important for my project.

**Comment by mmatvien** [comment]
+1

**Comment by drewhk** [comment]
Already done. Wait for the 1.1 release :)

**Comment by rkuhn** [comment]
GraphStage has been outfitted with these powers in #18605.
